src/main/java/io/nats/client/Options.java

```diff
@@ -1950,16 +1950,15 @@ public CharBuffer buildProtocolConnectOptionsString(String serverURI, boolean in
             try {
                 URI uri = this.createURIForServer(serverURI);
                 String userInfo = uri.getUserInfo();
-
                 if (userInfo != null) {
-                    String[] info = userInfo.split(":");
-
-                    if (info.length == 2) {
-                        uriUser = info[0];
-                        uriPass = info[1];
-                    } else {
+                    int at = userInfo.indexOf(":");

```

Comment:
I think that the split needs to happen _before_ the decode, because a Username or Token could have an encoded colon character in it `:`

consider the user `test:user` with the password `test:password`

that encoded string would be `test%3Auser:test%3Apassword`

Also, consider the token authentication with a token `test:token`

that encoded string would be `test%3Atoken`

Reply:
Yeah, the problem was the java uri `getUserInfo` call already decoded the entire part, so I would not know where to split. It's now fixed to properly handle encoded colon by using `getRawInfo` and decoding the parts separately,. Also made sure the plus `+` was not touched during the decoding.